web/modules/custom/nys_comment/nys_comment.module

```diff
@@ -175,3 +175,36 @@ function nys_comment_deploy_default_rejected_value() {
     $comment->save();
   }
 }
+
+/**
+ * Implements hook_cron().
+ */
+function nys_comment_cron() {
+  $content_types = ['bill'];
+  $current_session_year = \Drupal::config('nys_config.settings')->get('nys_session_year');
+
+  // Get all nodes with open comments created outside current session.
+  $query = Drupal::database()->select('node', 'n');
+  $query->join('node__field_ol_session', 'ns', 'n.nid = ns.entity_id');
+  $query->join('node__field_comments', 'nc', 'n.nid = nc.entity_id');
+  $query->fields('n', ['nid', 'type']);
+  $query->fields('ns', ['field_ol_session_value']);
+  $query->condition('nc.field_comments_status', '1', '!=');
```

Comment:
Search should limit itself to `nc.field_comments_status = 2`.  If someone has marked comments as hidden, it should stay that way until manually changed.